### ISSUE TITLE
ArchMemory x86_64 unmapPage bug

### ISSUE THREAD

**Issue by fkirc** [IAIK/sweb#76]
There seems to be a bug in sweb/arch/x86/64/source/ArchMemory.cpp. The unmapPage method does not free the page it is supposed to do.

```C
bool ArchMemory::unmapPage(uint64 virtual_page)
{
  ArchMemoryMapping m = resolveMapping(page_map_level_4_, virtual_page);

  assert(m.page_ppn != 0 && m.page_size == PAGE_SIZE);




------------------- Error at this line: ---------------------------------------------------------------
  PageManager::instance()->freePPN(m.pt_ppn, PAGE_SIZE); 
-------------------------- Should be changed to: -----------------------------------------------
 PageManager::instance()->freePPN(m.page_ppn, PAGE_SIZE); 




  bool empty = checkAndRemove<PageTableEntry>(getIdentAddressOfPPN(m.pt_ppn), m.pti);
  if (empty) 
  {
    PageManager::instance()->freePPN(m.pt_ppn, PAGE_SIZE);
    empty = checkAndRemove<PageDirPageTableEntry>(getIdentAddressOfPPN(m.pd_ppn), m.pdi);
  }
  if (empty) 
  {
    PageManager::instance()->freePPN(m.pd_ppn, PAGE_SIZE);
    empty = checkAndRemove<PageDirPointerTablePageDirEntry>(getIdentAddressOfPPN(m.pdpt_ppn), m.pdpti);
  }
  if (empty) 
  {
    PageManager::instance()->freePPN(m.pdpt_ppn, PAGE_SIZE);
    empty = checkAndRemove<PageMapLevel4Entry>(getIdentAddressOfPPN(m.pml4_ppn), m.pml4i);
  }
  return true;
}

```

**Comment by dgruss** [comment]
there is more broken right?
where is the present bit set to 0?

**Comment by fkirc** [comment]
Yes. Are these intentional bugs to be fixed by the students?

**Comment by dgruss** [comment]
On 28.04.2016 17:01, fkirc wrote:
> Yes. Are these intentional bugs to be fixed by the students?

No. But if you patch them and submit a patch here you will get points 
for it!

**Comment by fkirc** [comment]
Ok I will make a pull request soon

**Comment by dgruss** [comment]
solved in #80
